relayer/chain/substrate/writer.go

```diff
@@ -19,11 +19,12 @@ import (
 )
 
 type Writer struct {
-	conn     *Connection
-	messages <-chan chain.Message
-	headers  <-chan chain.Header
-	log      *logrus.Entry
-	nonce    uint32
+	conn         *Connection
+	messages     <-chan chain.Message
+	headers      <-chan chain.Header
+	log          *logrus.Entry
+	nonce        uint32
+	nonceCounter uint32
```

Comment:
`nonceCounter` looks to be a bit redundant. Can't we just increment `nonce` after every sending every transaction?